MWSE/TES3Inventory.cpp

```diff
@@ -234,8 +234,8 @@ namespace TES3 {
 
 	int Inventory::addItem_lua(sol::table params) {
 		TES3::MobileActor* mact = mwse::lua::getOptionalParamMobileActor(params, "mobile");
-		TES3::Item* item = mwse::lua::getOptionalParamObject<TES3::Item>(params, "item");
-		if (item == nullptr) {
+		TES3::PhysicalObject* item = mwse::lua::getOptionalParamObject<TES3::PhysicalObject>(params, "item");
+		if (item == nullptr || (!item->isItem() && item->objectType != TES3::ObjectType::LeveledItem)) {
```

Comment:
Here would be a welcome place to do the translation from leveled item to item. You could do:
```cpp
auto item = mwse::lua::getOptionalParamObject<TES3::PhysicalObject>(params, "item");
if (item == nullptr) {
	throw std::invalid_argument("tes3inventory:addItem: Invalid 'item' parameter provided.");
}

// Resolve leveled items.
if (item->objectType == ObjectType::LeveledItem) {
	item = static_cast<LeveledItem*>(item)->resolve();
	if (item == nullptr) {
		return;
	}
}

// Make sure the given parameter is actually an item.
if (!item->isItem()) {
	throw std::invalid_argument("tes3inventory:addItem: Invalid 'item' parameter provided.");
}
```